File: ui/src/app/edge/settings/network/getNetworkConfigRequest.ts
import { JsonrpcRequest } from "../../../shared/jsonrpc/base";

/**
 * Represents a JSON-RPC Request for 'getNetworkConfig'.
 *
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "method": "getNetworkConfig",
 *   "params": {}
 * }
 * </pre>
 */
export class GetNetworkConfigRequest extends JsonrpcRequest {

    private static METHOD: string = "getNetworkConfig";

    public constructor(
    ) {
        super(GET_NETWORK_CONFIG_REQUEST.METHOD, {});
    }

}

File: io.openems.edge.core/src/io/openems/edge/core/host/jsonrpc/GetNetworkConfigRequest.java
package io.openems.edge.core.host.jsonrpc;

import com.google.gson.JsonObject;

import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
import io.openems.common.exceptions.OpenemsException;
import io.openems.common.jsonrpc.base.JsonrpcRequest;

/**
 * Gets the current network configuration.
 *
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "method": "getNetworkConfig",
 *   "params": {}
 * }
 * </pre>
 */
public class GetNetworkConfigRequest extends JsonrpcRequest {

	public static final String METHOD = "getNetworkConfig";

	/**
	 * Parses a generic {@link JsonrpcRequest} to a {@link GetNetworkConfigRequest}.
	 *
	 * @param r the {@link JsonrpcRequest}
	 * @return the {@link GetNetworkConfigRequest}
	 * @throws OpenemsNamedException on error
	 */
	public static GetNetworkConfigRequest from(JsonrpcRequest r) throws OpenemsException {
		return new GetNetworkConfigRequest(r);
	}

	public GetNetworkConfigRequest() {
		super(METHOD);
	}

	private GetNetworkConfigRequest(JsonrpcRequest request) {
		super(request, METHOD);
	}

	@Override
	public JsonObject getParams() {
		return new JsonObject();
	}

}
